src/angulartics-piwik.js

```diff
@@ -118,7 +118,7 @@
 
                     if ($window._paq) {
                         $window._paq.push(['setDocumentTitle', $window.document.title]);
-                        $window._paq.push(['setCustomUrl', path]);
+                        $window._paq.push(['setCustomUrl',locationObj.$$absUrl]);
```

Comment:
I think, this should be optional und be customizable from parent component.

because in https://developer.matomo.org/guides/spa-tracking

matomo use this code-snippet for SPA

```
var currentUrl = location.href;
window.addEventListener('hashchange', function() {
    _paq.push(['setReferrerUrl', currentUrl]);
     currentUrl = '/' + window.location.hash.substr(1);
    _paq.push(['setCustomUrl', currentUrl]);
    _paq.push(['setDocumentTitle', 'My New Title']);

    // remove all previously assigned custom variables, requires Matomo (formerly Piwik) 3.0.2
    _paq.push(['deleteCustomVariables', 'page']); 
    _paq.push(['setGenerationTimeMs', 0]);
    _paq.push(['trackPageView']);

    // make Matomo aware of newly added content
    var content = document.getElementById('content');
    _paq.push(['MediaAnalytics::scanForMedia', content]);
    _paq.push(['FormAnalytics::scanForForms', content]);
    _paq.push(['trackContentImpressionsWithinNode', content]);
    _paq.push(['enableLinkTracking']);
});
```